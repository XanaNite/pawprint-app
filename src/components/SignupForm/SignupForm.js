import React, { Component } from 'react'

export default class SignupForm extends Component {

  render() {
    return (
      <form className='SignupForm'>
        <div className='email'>
          <label htmlfor="email">Email</label>
          <input type="text" name='email' id='email' />
        </div>
        <div className="password">
          <label for="password">Password</label>
          <input type="password" name='password' id='password' />
        </div>
        <button type='submit'>Submit</button>
      </form>
    )
  }
}
