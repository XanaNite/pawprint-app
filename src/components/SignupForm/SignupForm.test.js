import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SignupForm from './SignupForm';

describe(`SignupForm component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  };

  it('renders SignupForm by default', () => {
    const wrapper = shallow(<SignupForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('renders the SignupForm given props', () => {
    const wrapper = shallow(<SignupForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})
