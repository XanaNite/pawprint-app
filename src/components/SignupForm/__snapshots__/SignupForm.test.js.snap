// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm component renders SignupForm by default 1`] = `
<form
  className="SignupForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="name"
  >
    <label
      htmlFor="name"
    >
      Name
    </label>
    <input
      id="name"
      name="name"
      type="text"
    />
  </div>
  <div
    className="email"
  >
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      id="email"
      name="email"
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      id="password"
      name="password"
      type="password"
    />
  </div>
  <NavLink
    to="/login"
  >
    <button
      className="loginbuttons"
      type="button"
    >
      Login?
    </button>
  </NavLink>
  <button
    className="signUpButtons"
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`SignupForm component renders the SignupForm given props 1`] = `
<form
  className="SignupForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="name"
  >
    <label
      htmlFor="name"
    >
      Name
    </label>
    <input
      id="name"
      name="name"
      type="text"
    />
  </div>
  <div
    className="email"
  >
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      id="email"
      name="email"
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      id="password"
      name="password"
      type="password"
    />
  </div>
  <NavLink
    to="/login"
  >
    <button
      className="loginbuttons"
      type="button"
    >
      Login?
    </button>
  </NavLink>
  <button
    className="signUpButtons"
    type="submit"
  >
    Submit
  </button>
</form>
`;
